### PR TITLE
Limit how many local backups are kept

### LINKED ISSUE #5279: [IMPROVEMENT] Local Backups History Limit
### Please confirm these before moving forward.

- [x] I have searched for my feature proposal and have not found a work-in-progress/duplicate/resolved/discarded issue.
- [x] This improvement refers to an existing feature. If you want to suggest a new feature, please use [this template](https://github.com/Devolutions/UniGetUI/issues/new?labels=new-feature&projects=&template=feature-request.yml&title=%5BFEATURE+REQUEST%5D+%28Enter+your+description+here%29).
- [x] This improvement is not a bug. If you want to report a bug, please use [this template](https://github.com/Devolutions/UniGetUI/issues/new?labels=bug&projects=&template=bug-issue.yml&title=%5BBUG%5D+%28Enter+your+description+here%29).

### Describe the improvement

Currently, local backups are saved indefinitely. Add a number or date range indicating how many backups to keep.

### Describe how this improvement could help users

Prevents overloading disk space with older backups.

### PR DESCRIPTION
**Pull request by GabrielDuf**
This pull request introduces a comprehensive backup retention feature for local backups, allowing users to limit the number of backup files kept and to manage backup file naming and retention more flexibly. The changes affect both the backend logic and the user interface, and include robust tests to ensure correct behavior.

**Backup retention and file management enhancements:**

* Added new settings and UI controls to allow users to specify the maximum number of local backups to retain, with options for predefined limits or a custom value. The oldest backups are deleted when the limit is exceeded. 
* Updated backup file naming logic to support timestamped files and to only apply retention logic when separate files per backup (timestamping) is enabled.
* Added new settings keys (`MaxLocalBackupCount`, `MaxLocalBackupCountCustom`) and their resolution in the settings engine.

**User interface improvements:**

* Updated the backup settings page to include new controls for backup retention, including a combobox for selecting retention limits and a textbox for custom values, with dynamic enablement and layout. 
* Improved the search index for settings to include new backup retention options and updated the description of timestamped backups.

**Testing and reliability:**

* Added a comprehensive test suite (`LocalBackupManagerTests.cs`) to verify backup retention logic, file naming, settings integration, and edge cases such as unrelated files and non-Gregorian calendars.

These changes provide users with greater control over backup file management, improve clarity in the UI, and ensure robust, predictable behavior for backup retention.

### CODE REVIEW

File: src/UniGetUI.Core.Tools/LocalBackupManager.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+    public static string BuildFileName() => BuildFileName(DateTime.Now);
+
+    public static string BuildFileName(DateTime timestamp)
+    {
+        string fileName = ResolveFileNameBase();
+        if (Settings.Get(Settings.K.EnableBackupTimestamping))
+            fileName += " " + timestamp.ToString(TimestampFormat, CultureInfo.InvariantCulture);
+
+        return fileName + BackupExtension;
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Core.Tools/LocalBackupManager.cs:53]: fixed in [0a66579](https://github.com/Devolutions/UniGetUI/pull/5318/commits/0a665795a25bbabce014ba2883ec7b64fb019b9f)

File: src/UniGetUI.Core.Tools/LocalBackupManager.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            return 0;
+
+        List<(DateTime Timestamp, string Path)> backups = [];
+        foreach (string path in Directory.EnumerateFiles(
+            directory,
+            "*" + BackupExtension,
+            SearchOption.TopDirectoryOnly))
+        {
+            if (GetBackupTimestamp(Path.GetFileName(path), fileNameBase) is { } timestamp)
+                backups.Add((timestamp, path));
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Core.Tools/LocalBackupManager.cs]: fixed in [0a66579](https://github.com/Devolutions/UniGetUI/pull/5318/commits/0a665795a25bbabce014ba2883ec7b64fb019b9f)

File: src/UniGetUI.Core.Tools/LocalBackupManager.cs [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+                {
+                }
+            }
+            catch (IOException) when (File.Exists(path))
+            {
+                continue;
+            }
+
+            await File.WriteAllTextAsync(path, contents);
+            return path;
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Core.Tools/LocalBackupManager.cs]: [a696460](https://github.com/Devolutions/UniGetUI/pull/5318/commits/a69646026add415a50c2a05581f381478c063431)

File: src/UniGetUI.Core.Tools/LocalBackupManager.cs [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+            value = value[..separator];
+        }
+
+        if (TryParseTimestamp(value, CultureInfo.InvariantCulture, out DateTime timestamp))
+            return (timestamp, sequence);
+
+        return TryParseTimestamp(value, CultureInfo.CurrentCulture, out timestamp)
+            ? (timestamp, sequence)
+            : null;
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Core.Tools/LocalBackupManager.cs]: [a696460](https://github.com/Devolutions/UniGetUI/pull/5318/commits/a69646026add415a50c2a05581f381478c063431)

**Review by mamoreau-devolutions — APPROVED** [review]
